erdgo/blockchain/proxy.go

```diff
@@ -25,13 +25,23 @@ const (
 	getNetworkStatusEndpoint     = "network/status/%v"
 )
 
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+var (
+	Client HTTPClient
```

Comment:
Please do not use exported (static) instances. This is a bad practice as it is incredibly hard to test the instances that use exported variables.

Reply:
I've actually did in order to be able to write a test, the 2 methods `get` and `post` where using client.Default that made it impossible to test.

As an alternative I could inject in the constructor and don't export it, does this sound better?
(will actually have to make a new constructor so I don't break existing clients)